rpc/src/v1/impls/parity.rs

```diff
@@ -449,4 +452,34 @@ impl<C, M, U, S> Parity for ParityClient<C, M, U> where
 				.map(|res| res.into_iter().map(|res| res.output.into()).collect())
 				.map_err(errors::call)
 	}
+
+	fn submit_work_detail(&self, nonce: RpcH64, pow_hash: RpcH256, mix_hash: RpcH256) -> Result<SubmitDetailResult> {
+		// TODO [ToDr] Should disallow submissions in case of PoA?
+		let nonce: EthH64 = nonce.into();
+		let pow_hash: EthH256 = pow_hash.into();
+		let mix_hash: EthH256 = mix_hash.into();
+		trace!(target: "miner", "submit_work_detail: Decoded: nonce={}, pow_hash={}, mix_hash={}", nonce, pow_hash, mix_hash);
+
+		let seal = vec![rlp::encode(&mix_hash).into_vec(), rlp::encode(&nonce).into_vec()];
+		let import = self.miner.submit_seal(pow_hash, seal)
+			.and_then(|block| self.client.import_sealed_block(block));
+
+		match import {
```

Comment:
JSONRPC supports errors, so there's no need for `SubmitDetailResult` type. Add a new error helper in `rpc/src/v1/helpers/errors.rs`, and then change this function's return type to `Result<H256>`. You can then change the return value to something like `import.map_err(errors::submit_work)`.

Reply:
I have completed the change.